biofefi/machine_learning/nn_models.py

```diff
@@ -0,0 +1,271 @@
+                )
+
+    def predict(self, X, return_probs=False) -> np.ndarray:
+        """
+        Predict the target values using the trained BRNN Regressor.
+
+        Args:
+            X (np.ndarray): The input data.
+            return_probs (bool): Whether to return the predicted
+
+        Returns:
+            np.ndarray: The predicted target values.
+
+        Raises:
+            ValueError: If an error occurs during prediction.
+        """
+        if isinstance(X, pd.DataFrame):
+            X = X.to_numpy()
+        X = torch.tensor(X, dtype=torch.float32).to(self.device)
+
+        try:
+            self.eval()
+            with torch.no_grad():
+                outputs = self(X)
+
+                if outputs.size(1) == 1:  # Binary classification
+                    probabilities = torch.sigmoid(outputs).cpu().numpy()
+                    return (
+                        probabilities
+                        if return_probs
+                        else (probabilities > BrnnOptions.classification_cutoff).astype(
+                            int
+                        )
+                    )
+
+                else:  # Multi-class classification
+                    probabilities = torch.softmax(outputs, dim=1).cpu().numpy()
+                    return (
+                        probabilities
+                        if return_probs
+                        else np.argmax(probabilities, axis=1)
+                    )
+
+        except Exception as e:
+            raise ValueError(f"Error occured during prediction of BRNN Classifier: {e}")
+
+
+class BayesianRegularisedNNRegressor(BaseNetwork, BaseEstimator, RegressorMixin):
+    """
+    This class defines a Bayesian Regularized Neural
+    Network for regression tasks.
+
+    Args:
+        problem_type (ProblemTypes): The type of problem
+        (Regression).
+        - layer1: The first linear layer of the network.
+        - layer2: The second linear layer of the network.
+        - output_layer: The output linear layer of the network.
+    """
+
+    def __init__(self, problem_type: ProblemTypes):
+        """
+        Initializes the BayesianRegularisedNNRegressor class.
+        """
+        super().__init__()
+        self.problem_type = problem_type
+        self._name = ModelNames.BRNNRegressor
+        self.layer1, self.layer2, self.output_layer = None, None, None
+
+    def _initialize_network(self, input_dim, output_dim):
+        """
+        Initializes the network layers for BRNN regression.
+
+        Args:
+            input_dim (int): The input dimension of the data.
+            output_dim (int): The output dimension of the
+            data, determined dynamically.
+        """
+        # Define hidden layers and output layer
+        self.layer1 = nn.Linear(input_dim, BrnnOptions.hidden_dim)
+        self.layer2 = nn.Linear(BrnnOptions.hidden_dim, BrnnOptions.hidden_dim)
+        self.output_layer = nn.Linear(BrnnOptions.hidden_dim, output_dim)
```

Comment:
Add a BrnnOptions instance to the constructor of this class and use that. BrnnOptions is a class

Reply:
Ok